builder/virtualbox/common/driver_4_2.go

```diff
@@ -24,9 +24,11 @@ func (d *VBox42Driver) CreateSATAController(vmName string, name string, portcoun
 		return err
 	}
 
-	portCountArg := "--sataportcount"
-	if strings.HasPrefix(version, "4.3") || strings.HasPrefix(version, "5.") || strings.HasPrefix(version, "6.") {
-		portCountArg = "--portcount"
+	portCountArg = "--portcount"
```

Comment:
Woops, here you should create `portCountArg`
```suggestion
	portCountArg := "--portcount"
```